src/Shipments/PriceCalculator.php

```diff
@@ -66,7 +69,11 @@ public function calculateOptionsPrice(ShipmentInterface $shipment, ServiceRateIn
                 );
             }
 
-            $price += $prices[$option->getId()];
+            $optionPrice = $prices[$option->getId()];
+
+            if ($optionPrice !== null) {
+                $price += $optionPrice;
+            }
```

Comment:
You will never return `null` now, only `0`. You could add an early return here instead of the current check:
```
if ($optionPrice === null) {
  return null;
}

$price += $optionPrice;
```